rollout/experiment.go

```diff
@@ -64,7 +64,9 @@ func GetExperimentFromTemplate(r *v1alpha1.Rollout, stableRS, newRS *appsv1.Repl
 			Name:     templateStep.Name,
 			Replicas: templateStep.Replicas,
 		}
-		template.Service = &v1alpha1.TemplateService{}
+		if templateStep.Weight != nil {
```

Comment:
```suggestion
		if (templateStep.Weight != nil || templateStep.Service != nil) {
```
should we put this already in?

atm the spec doesn't allow `templateStep.Service` to be anything but:
- nil
- empty object

so this shouldn't be a breaking change (famous last words)

Reply:
see: https://github.com/alexef/argo-rollouts/blob/master/pkg/apis/rollouts/v1alpha1/experiment_types.go#L92

Reply:
nevermind, this is not possible without a Rollout spec change.